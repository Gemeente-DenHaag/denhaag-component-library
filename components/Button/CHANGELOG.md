# @gemeente-denhaag/button

## 0.2.3

### Patch Changes

- Updated dependencies [281fa34]
  - @gemeente-denhaag/icons@1.0.0
