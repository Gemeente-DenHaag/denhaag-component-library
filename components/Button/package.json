{
  "name": "@gemeente-denhaag/button",
  "description": "A Button component",
  "author": "Municipality of The Hague",
  "license": "EUPL-1.2",
  "version": "0.2.3-alpha.48",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "exports": "./dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c ../../rollup.config.js",
    "clean": "yarn rimraf dist tsconfig.tsbuildinfo"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/nl-design-system/denhaag.git",
    "directory": "components/Button"
  },
  "bugs": "https://github.com/nl-design-system/denhaag/issues",
  "dependencies": {
    "@gemeente-denhaag/baseprops": "^0.2.3-alpha.48",
    "@material-ui/core": "^4.11.0"
  },
  "devDependencies": {
    "@gemeente-denhaag/icons": "^0.2.3-alpha.48"
  },
  "peerDependencies": {
    "react": "^17.0.1"
  },
  "gitHead": "dcf72a9b79266c1ebede35aff4a02dd9121a980f"
}
