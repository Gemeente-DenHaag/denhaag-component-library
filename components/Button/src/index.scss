.denhaag-button {
  --denhaag-button-focus-border-offset: calc(-1 * var(--denhaag-focus-border-width));

  align-items: center;
  background-color: var(--denhaag-button-primary-action-background-color);
  border: 0;
  border-radius: var(--denhaag-button-border-radius);
  color: var(--denhaag-button-primary-action-color);
  cursor: var(--denhaag-button-cursor, default);
  display: inline-flex;
  font-family: var(--denhaag-button-font-family);
  font-weight: var(--denhaag-button-font-weight, 400);
  font-size: var(--denhaag-button-font-size, var(--denhaag-typography-scale-base-font-size));
  justify-content: center;
  line-height: 1.5;
  padding-block-end: var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block));
  padding-block-start: var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block));
  padding-inline-end: var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline));
  padding-inline-start: var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline));
  position: relative;
  text-decoration: none;
  transition-property: background-color, border-color, color, transform;
  transition-duration: var(--denhaag-button-transition-duration);
  transition-timing-function: var(--denhaag-button-transition-timing-function);
  transition-delay: var(--denhaag-button-transition-delay);
  transform: translate(var(--denhaag-button-transform-x, 0), var(--denhaag-button-transform-y, 0));

  @media (prefers-reduced-motion: reduce) {
    --denhaag-button-transition-duration: 0;
  }
}

.denhaag-button--reduced-button-transition {
  --denhaag-button-transition-duration: 0;
}

a.denhaag-button {
  --denhaag-button-cursor: pointer;
}

.denhaag-button.denhaag-button--hover,
.denhaag-button:hover {
  --denhaag-button-transform-y: var(--denhaag-button-hover-transform-y);

  background-color: var(--denhaag-button-primary-action-hover-background-color);
  color: var(--denhaag-button-primary-action-hover-color);
}

.denhaag-button--focus-visible::after,
.denhaag-button:focus-visible::after {
  border: var(--denhaag-focus-border);
  border-radius: var(--denhaag-border-radius);
  bottom: var(--denhaag-button-focus-border-offset);
  content: "";
  display: block;
  left: var(--denhaag-button-focus-border-offset);
  position: absolute;
  right: var(--denhaag-button-focus-border-offset);
  top: var(--denhaag-button-focus-border-offset);
}

.denhaag-button--focus,
.denhaag-button:focus,
.denhaag-button--focus-visible,
.denhaag-button:focus-visible {
  --denhaag-button-transform-y: var(--denhaag-button-focus-transform-y);

  outline: none;
}

.denhaag-button.denhaag-button--disabled,
.denhaag-button.denhaag-button:disabled {
  background-color: var(--denhaag-button-primary-action-disabled-background-color);
  color: var(--denhaag-button-primary-action-disabled-color);
}

.denhaag-button--secondary-action {
  // Due to the accessibility focus outline overlapping with the border, the outline needs to be repositioned for the border variant.
  --denhaag-button-focus-border-offset: calc(-1 * (var(--denhaag-focus-border-width) * 1.5));

  background-color: var(--denhaag-button-secondary-action-background-color);
  border-color: var(--denhaag-button-secondary-action-border-color);
  border-style: solid;
  border-width: var(--denhaag-button-border-width);
  color: var(--denhaag-button-secondary-action-color);
  padding-block-end: calc(
    var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block)) -
      var(--denhaag-button-border-width)
  );
  padding-block-start: calc(
    var(--denhaag-button-medium-size-padding-block, var(--denhaag-button-padding-block)) -
      var(--denhaag-button-border-width)
  );
  padding-inline-end: calc(
    var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline)) -
      var(--denhaag-button-border-width)
  );
  padding-inline-start: calc(
    var(--denhaag-button-medium-size-padding-inline, var(--denhaag-button-padding-inline)) -
      var(--denhaag-button-border-width)
  );
}

.denhaag-button--secondary-action.denhaag-button--hover,
.denhaag-button--secondary-action:hover {
  background-color: var(--denhaag-button-secondary-action-hover-background-color);
  border-color: var(--denhaag-button-secondary-action-hover-border-color);
  color: var(--denhaag-button-secondary-action-hover-color);
}

.denhaag-button--secondary-action.denhaag-button--disabled,
.denhaag-button--secondary-action.denhaag-button:disabled {
  background-color: var(--denhaag-button-secondary-action-disabled-background-color);
  border-color: var(--denhaag-button-secondary-action-disabled-border-color);
  color: var(--denhaag-button-secondary-action-disabled-color);
}

.denhaag-button--large {
  --denhaag-button-font-size: var(--denhaag-typography-scale-lg-font-size);
  --denhaag-button-medium-size-padding-block: var(--denhaag-button-large-size-padding-block);
  --denhaag-button-medium-size-padding-inline: var(--denhaag-button-large-size-padding-inline);

  line-height: var(--denhaag-button-large-size-line-height);
}

.denhaag-button--large.denhaag-button--secondary-action {
  --denhaag-button-medium-size-padding-block: calc(
    var(--denhaag-button-large-size-padding-block, var(--denhaag-button-padding-block)) -
      var(--denhaag-button-border-width)
  );
  --denhaag-button-medium-size-padding-inline: calc(
    var(--denhaag-button-large-size-padding-inline, var(--denhaag-button-padding-inline)) -
      var(--denhaag-button-border-width)
  );
}

.denhaag-button__icon > *:first-child {
  font-size: var(--denhaag-typography-scale-lg-font-size);
}

.denhaag-button--large .denhaag-button__icon > *:first-child {
  font-size: var(--denhaag-typography-scale-xl-font-size);
}

.denhaag-button--start-icon .denhaag-button__icon {
  display: inherit;
  margin-inline-end: 8px;
  margin-inline-start: -4px;
}

.denhaag-button--end-icon .denhaag-button__icon {
  display: inherit;
  margin-inline-end: -4px;
  margin-inline-start: 8px;
}

.denhaag-button__sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding-block-start: 0;
  padding-block-end: 0;
  padding-inline-start: 0;
  padding-inline-end: 0;
  margin-block-start: -1px;
  margin-block-end: -1px;
  margin-inline-start: -1px;
  margin-inline-end: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
}

.denhaag-button--icon-only {
  --denhaag-button-medium-size-padding-inline: 0.75rem;

  height: var(--denhaag-button-icon-only-height);
}

.denhaag-button--icon-only .denhaag-button__icon {
  height: var(--denhaag-button-icon-only-icon-height);
}
