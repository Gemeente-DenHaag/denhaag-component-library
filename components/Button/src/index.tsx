import React from 'react';
import BaseProps from '@gemeente-denhaag/baseprops';
import './button.css';
import clsx from 'clsx';

export interface ButtonProps extends BaseProps {
  /**
   * Simple click handler
   */
  onClick?: (event: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>) => void;

  /**
   * Size of the component
   */
  size?: 'medium' | 'large';

  /**
   * Disables Button
   */
  disabled?: boolean;

  /**
   * Button variant
   */
  variant?: 'primary-action' | 'secondary-action';

  /**
   * HTML type prop
   */
  type?: 'button' | 'submit' | 'reset';

  /**
   * Icon placed at the start of the button
   */
  startIcon?: React.ReactNode;

  /**
   * Icon placed at the end of the button
   */
  endIcon?: React.ReactNode;
}

/**
 * Primary UI component for user interaction
 */
export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const sizeClass = `denhaag-button--${props.size ?? 'medium'}`;
  const variantClass = `denhaag-button--${props.variant ?? 'primary-action'}`;
  const rootClassNames = clsx('denhaag-button', sizeClass, variantClass, props.className);

  const iconClassNames = clsx(
    {
      'denhaag-button__start-icon': props.startIcon !== undefined,
      'denhaag-button__end-icon': props.endIcon !== undefined,
    },
    `denhaag-button__icon-size-${props.size ?? 'medium'}`,
  );
  const iconWrapped = <span className={iconClassNames}>{props.startIcon || props.endIcon}</span>;

  return (
    <button className={rootClassNames} onClick={props.onClick} disabled={props.disabled} type={props.type}>
      {props.startIcon !== undefined ? iconWrapped : ''}
      {props.children}
      {props.endIcon !== undefined ? iconWrapped : ''}
    </button>
  );
};

/**
 * Default export for Button
 */
export default Button;
