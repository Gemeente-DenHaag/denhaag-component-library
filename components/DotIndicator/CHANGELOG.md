# @gemeente-denhaag/dotindicator

## 0.1.1

### Patch Changes

- @gemeente-denhaag/button@0.2.3
