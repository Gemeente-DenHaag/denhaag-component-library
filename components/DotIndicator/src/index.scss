.denhaag-dot-indicator {
  position: relative;
  display: inline-flex;
  flex-shrink: 0;
  vertical-align: middle;
}

.denhaag-dot-indicator__dot {
  position: absolute;
  background-color: var(--denhaag-dot-indicator-background-color, var(--denhaag-color-red-3));
  width: var(--denhaag-dot-indicator-size, 12px);
  height: var(--denhaag-dot-indicator-size, 12px);
  transform: scale(1) translate(50%, -50%);
  border-radius: 50%;
  border: 2px solid var(--denhaag-color-white);
}

.denhaag-dot-indicator--overlap-rectangle .denhaag-dot-indicator__dot {
  transform: scale(1) translate(50%, -50%);
  top: 0;
  right: 0;
}

.denhaag-dot-indicator--overlap-circle .denhaag-dot-indicator__dot {
  transform: scale(1) translate(50%, -50%);
  top: 14%;
  right: 14%;
}
