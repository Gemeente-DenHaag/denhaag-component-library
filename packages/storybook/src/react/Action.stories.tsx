import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { ActionSingle, ActionSingleProps } from '@gemeente-denhaag/action';
import { FormatDateLabels } from '@gemeente-denhaag/utils';

import readme from '../../../../components/Action/README.md';

const labels: FormatDateLabels = {
  today: 'vandaag',
  yesterday: 'gisteren',
  before: 'vóór',
  approachingDeadline: (daysDifference: number) => {
    if (daysDifference === 1) {
      return `nog ${daysDifference} dag`;
    }
    return `nog ${daysDifference} dagen`;
  },
};

const exampleArgs: ActionSingleProps = {
  children: <strong>Taak</strong>,
  link: '#example',
  labels,
};

const meta = {
  id: 'react-action',
  title: 'React/Actions/Action',
  component: ActionSingle,
  args: exampleArgs,
  tags: ['autodocs'],
  parameters: {
    chromatic: { viewports: [1768, 1280, 768, 360] },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as Meta<typeof ActionSingle>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const Hover: Story = {
  args: { ...Default.args, className: 'denhaag-action--hover' },
};

export const Focus: Story = {
  args: { ...Default.args, className: 'denhaag-action--focus' },
};

export const WithDate: Story = {
  args: { ...Default.args, dateTime: '2023-09-28T19:47:36.593Z' },
};

export const WithDateRelative: Story = {
  args: { ...Default.args, dateTime: '2023-10-02T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

export const WithDateWarning: Story = {
  args: { ...Default.args, dateTime: '2023-09-30T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

// ### Multiple - hover

// <Canvas>
//   <Story name="Multiple - hover">
//     {() => (
//       <Action
//         actions={
//           <ButtonLink href="#example" className="utrecht-button-link--hover">
//             Actie
//           </ButtonLink>
//         }
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - focus

// <Canvas>
//   <Story name="Multiple - focus">
//     {() => (
//       <Action
//         actions={
//           <ButtonLink href="#example" className="utrecht-button-link--focus">
//             Actie
//           </ButtonLink>
//         }
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with date

// <Canvas>
//   <Story name="Multiple - with date">
//     {() => (
//       <Action
//         dateTime={"2023-09-28T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with date - relative

// <Canvas>
//   <Story name="Multiple - with date - relative">
//     {() => (
//       <Action
//         relativeDate
//         dateTime={"2023-10-02T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with warning date - relative

// <Canvas>
//   <Story name="Multiple - with warning date - relative">
//     {() => (
//       <Action
//         relativeDate
//         dateTime={"2023-09-30T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>
