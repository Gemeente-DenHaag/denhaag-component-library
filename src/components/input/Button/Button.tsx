import MaterialButton from '@material-ui/core/Button'
import React from 'react'
import BaseProps from '../../BaseProps/BaseProps'
import './button-design-tokens.module.css'
// import { createMuiTheme, Theme, ThemeProvider } from "@material-ui/core";

export interface ButtonProps extends BaseProps {
  /**
   * Simple click handler
   */
  onClick?: () => void

  /**
   * Color for the component
   */
  color?: 'primary' | 'secondary' | 'default'

  /**
   * The url to link to when the button is clicked.
   * If defined, an a element will be used as the root node.
   */
  href?: string

  /**
   * Size of the component
   */
  size?: 'small' | 'medium' | 'large'

  /**
   * Disables Button
   */
  disabled?: boolean

  /**
   * Button variant
   */
  variant?: 'outlined' | 'contained' | 'text'
}

/**
 * Primary UI component for user interaction
 */
export const Button: React.FC<ButtonProps> = (
  props: ButtonProps
) => {
  return (
      <MaterialButton {...props}>{props.children}</MaterialButton>
  )
}

/**
 * Default export for Button
 */
export default Button
