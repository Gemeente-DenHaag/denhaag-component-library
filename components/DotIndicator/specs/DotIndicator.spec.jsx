import * as React from 'react';
import DotIndicator from '../src';
import Button from '../../Button';
import Avatar from '../../Avatar';

describe(`DotIndicator tests`, () => {
  it('a11y and snapshots', () => {
    cy.snapshots(
      <DotIndicator overlap={'rectangle'}>
        <Button variant={'secondary'}>Read messages</Button>
      </DotIndicator>,
      {},
    );
    cy.snapshots(
      <DotIndicator overlap={'circle'}>
        <Avatar></Avatar>
      </DotIndicator>,
      {},
    );
  });
});
