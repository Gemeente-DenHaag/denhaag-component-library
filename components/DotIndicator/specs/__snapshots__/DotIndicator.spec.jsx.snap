exports[`DotIndicator tests > a11y and snapshots #0`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <span class="denhaag-dot-indicator denhaag-dot-indicator--overlap-rectangle"
      ><button variant="secondary" type="button" class="denhaag-button">
        <span class="denhaag-button__label">Read messages</span></button
      ><span class="denhaag-dot-indicator__dot"></span
    ></span>
  </div>
</div>
`;

exports[`DotIndicator tests > a11y and snapshots #1`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <span class="denhaag-dot-indicator denhaag-dot-indicator--overlap-circle"
      ><div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault">
        <svg
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
          aria-hidden="true"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          ></path>
        </svg>
      </div>
      <span class="denhaag-dot-indicator__dot"></span
    ></span>
  </div>
</div>
`;
